crates/secrets-nats-kv/src/client.rs

```diff
@@ -52,7 +52,7 @@ pub async fn put_secret(
         .expect("should be able to encrypt the secret");
     nats_client
         .request_with_headers(
-            format!("{subject_base}.v0.{}.put_secret", secret.name),
+            format!("{subject_base}.v0.nats-kv.put_secret"),
```

Comment:
I'm a little bit worried about putting more into `core` just because this function is specific to the nats-kv backend, but I think we could make the argument for sure for the secrets-client 🤔

Reply:
AH -- yeah you're right, it definitely shouldn't go in `core` since it's use dthere, but at least inside the crate it should probably be somewhere else